### PR TITLE
feat: add cfg reward rate event handler

### PR DESCRIPTION
**Pull request by offerijns**

### CODE REVIEW

File: src/domain/AOReward.ts [hunk 1]
```diff
@@ -62,17 +63,39 @@ export function calculateAORewards(date: BigInt, pool: Pool): void {
...
-  log.debug('setting system AO rewards rate, toDateAORewardAggregateValue {}', [
-    systemRewards.toDateAORewardAggregateValue.toString(),
-  ])
-  if (systemRewards.toDateAORewardAggregateValue.lt(BigDecimal.fromString(aoRewardsCeiling))) {
-    systemRewards.aoRewardRate = BigDecimal.fromString('0.0017')
-    systemRewards.save()
+function getAORewardRate(systemRewards: RewardDayTotal): BigDecimal {
+  let defaultRewardRate = BigDecimal.fromString('0.0017')
+
+  let isBelowLimit = systemRewards.toDateRewardAggregateValue.lt(BigDecimal.fromString(aoRewardsCeiling))
```
Comment (philipstanislaus) [inline src/domain/AOReward.ts:69]: Did you try logging this value? Could well be that there is something wrong with decimals here.

File: src/domain/Reward.ts [hunk 2]
```diff
@@ -143,17 +144,41 @@ export function calculateRewards(date: BigInt, pool: Pool): void {
...
-  log.debug('setting system rewards rate, toDateRewardAggregateValue {}', [
-    systemRewards.toDateRewardAggregateValue.toString(),
-  ])
-  if (systemRewards.toDateRewardAggregateValue.lt(BigDecimal.fromString(rewardsCeiling))) {
-    systemRewards.rewardRate = BigDecimal.fromString('0.0042')
-    systemRewards.save()
+function getInvestorRewardRate(systemRewards: RewardDayTotal): BigDecimal {
+  let defaultRewardRate = BigDecimal.fromString('0.0042')
+
+  let isBelowLimit = systemRewards.toDateRewardAggregateValue.lt(BigDecimal.fromString(rewardsCeiling))
```
Comment (philipstanislaus) [inline src/domain/Reward.ts:150]: Likewise

**Review by jpangelle — APPROVED** [review]
👍

File: src/domain/AOReward.ts [hunk 3]
```diff
@@ -65,7 +65,6 @@ export function calculateAORewards(date: BigInt, pool: Pool): void {
 
 function getAORewardRate(systemRewards: RewardDayTotal): BigDecimal {
   let defaultRewardRateBelowLimit = BigDecimal.fromString('0.0017')
-  let defaultRewardRateAboveLimit = BigDecimal.fromString('0.0003')
```
Comment (jpangelle) [inline src/domain/AOReward.ts]: Not sure we still needed this logic. If so, I can add it back.

**Pull request closed without merging**